test/common/config/metadata_test.cc

```diff
@@ -49,6 +49,57 @@ TEST(MetadataTest, MetadataValuePath) {
             ProtobufWkt::Value::KindCase::KIND_NOT_SET);
 }
 
+// Test case for typed metadata: original Any data in metadata field
+TEST(MetadataTest, OriginalTypedMetadataValue) {
```

Comment:
I don't feel right about these two unit tests (OriginalTypedMetadataValue and PackedTypedMetadataValue). It seems to me that they are testing the behavior of the APIs of Protobufs (i.e. `Any` and `envoy::config::core::v3::Metadata`) . I think we should instead test the `TypedMetadataImpl` constructor (which covers the TypedMetadataImpl::populateFrom function changed in this PR)

Reply:
Yes, this test does not look to test any Envoy specific behavior. Adding TypedMetadataTest tests is more relevant.

Reply:
Make sense. They are removed.